multi_company_holding_invoicing/models/invoice.py

```diff
@@ -1,73 +1,53 @@
 # -*- coding: utf-8 -*-
-# © 2015 Akretion (http://www.akretion.com)
-# Sébastien BEAU <sebastien.beau@akretion.com>
-# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
-
-
-from openerp import models, fields, api
-from openerp.tools.translate import _
-from openerp.exceptions import Warning as UserError
-from openerp.addons.connector.session import ConnectorSession
-from openerp.addons.connector.queue.job import job
+from odoo import models, fields, api, _
+from odoo.exceptions import UserError
+from odoo.addons.queue_job.job import job
 import logging
 _logger = logging.getLogger(__name__)
 
 
-@job
-def generate_child_invoice_job(session, model_name, args):
-    invoice = session.env['account.invoice'].browse(args.get('invoice_id'))
-    domain = [
-        ('company_id', '=', args.get('company_id')),
-        ('id', 'in', invoice.holding_sale_ids.ids)
-    ]
-    child_invoices = session.env['child.invoicing']._generate_invoice(domain)
-    child_invoices.write({'holding_invoice_id': args.get('invoice_id')})
-    for child_invoice in child_invoices:
-        child_invoice.signal_workflow('invoice_open')
-    return True
-
-
 class AccountInvoice(models.Model):
     _inherit = 'account.invoice'
 
     holding_sale_ids = fields.One2many('sale.order', 'holding_invoice_id')
     holding_sale_count = fields.Integer(
         compute='_compute_holding_sale_count',
-        string='# of Sales Order',
+        string=u'# of Sales Order',
         compute_sudo=True)
     sale_count = fields.Integer(
         compute='_compute_sale_count',
         string='# of Sales Order',
         compute_sudo=True)
+    holding_invoice_id = fields.Many2one('account.invoice', 'Holding Invoice')
     child_invoice_ids = fields.One2many(
         'account.invoice', 'holding_invoice_id')
     child_invoice_count = fields.Integer(
         compute='_compute_child_invoice_count',
-        string='# of Invoice',
+        string=u'# of Invoice',
         compute_sudo=True)
-    holding_invoice_id = fields.Many2one('account.invoice', 'Holding Invoice')
     child_invoice_job_ids = fields.One2many('queue.job', 'holding_invoice_id')
     child_invoice_job_count = fields.Integer(
         compute='_compute_child_invoice_job_count',
-        string='# of Child Invoice Jobs',
+        string=u'# of Child Invoice Jobs',
         compute_sudo=True)
 
-    @api.multi
     def _compute_holding_sale_count(self):
         for inv in self:
             inv.holding_sale_count = len(inv.holding_sale_ids)
 
-    @api.multi
     def _compute_sale_count(self):
         for inv in self:
-            inv.sale_count = len(inv.sale_ids)
+            sale_ids = []
```

Comment:
I think we have a perf issue here.
Indeed for counting the number of sale order we are reading in the database all the sale order line and all the sale order (huge select!).